open_bucket_hashtable.h

```diff
@@ -203,10 +199,12 @@ template <typename GID, typename LID, int maxBucketOverflow = 4, GID EMPTYBUCKET
       }
 
       bool operator==(iterator other) const {
-         return &hashtable->buckets[index] == &other.hashtable->buckets[other.index];
+         // comparison of iterators between two different hashtables undefined
+         assert(hashtable == other.hashtable);
```

Comment:
`assert` does nothing without debug flags on, including the check itself.

Reply:
But shouldn't this simply return false if the hashtables mismatch? Because then they are not equal.

Reply:
STL has comparisons between iterators of different collections undefined, due to implementing `<=>` i  suppose. Comparing iterators that can't reach each other is nonsense rather than true or false.